dbms/src/Storages/KVStore/MultiRaft/RegionCFDataBase.h

```diff
@@ -64,22 +65,35 @@ enum class DupCheck
     AllowSame,
 };
 
+
 template <typename Trait>
-struct RegionCFDataBase
+struct DeletedKeysHolder
+{
+};
+
+template <>
+struct DeletedKeysHolder<RegionLockCFDataTrait>
+{
+    std::vector<std::shared_ptr<const TiKVKey>> deleted_keys;
```

Comment:
Could we just record the keys in `handle_by_index_func`? For example, if we found it is a lock, we could record this to the holder, and then apply it after `RegionTable::writeCommittedByRegion` and before `meta.setApplied(index, term);`?